mantidproject/mantid
# Integrate QResolution into Sans Gui

Fixes #13314 
## For Testing:

This is an effort to integrate the QResolution effort inot the SANS Reduction workflow.

The test files can be found here: \olympic\Babylon5\Scratch\Anton\Testing\13312_Integrate_Q1D_into_GUI

The folder also contains the installer of the last Jenkins build.

Make sure that the files are in your Mantid directories.

There are a whole bunch of user files:
- USER_FILE_NO_QRESOLUTION.txt: without QResolution settings
- USER_FILE_QRESOLUTION_OFF.txt: QResolution turned actively off
- USER_FILE_QRESOLUTION_CIRCULAR.txt : QResolution with circular aperature
- USER_FILE_QRESOLUTION_RECTANGULAR.txt: QResolution with rectangular aperture
- USER_FILE_QRESOLUTION_INCOMPLETE.txt: An incomplete QResolution setting
- USER_FILE_QRESOLUTION_BAD_MODERATOR.txt: QResolution with invalid moderator file
- USER_FILE_QRESOLUTION_PART.txt: QResolution with missing information

**Test GUI logic**
1. Start ISIS GUI and set the isntrument to SANS2DTUBE
2. Load the user file USER_FILE_NO_QRESOLUTION.txt
   - Confirm that the QResolution checkbox on the Reduction Settigns tab is disabled
3. Load the data set SANS2D00028797.nxs as the sample data set
4. Enable the QResolution checkbox
   - Confirm that "Circ." is selected
   - Confirm that A1 and A2 are enabled but W1 and W2 are disabled
   - Confirm that LCollim and Delta R are enabled
   - Confirm that the Moderator file path is not enabled (it never is; you have to load it via the user file)
5. Run the 1D Reduce
   - Confirm that a Runtime Error is thrown with information regarding a missing moderator file. 
6. Go back to the Reduction Settings tab and switch from Circ to Rect.
   - Confirm that A1 and A2 switches to H1 and H2
   - Confirm that W1 and W2 are now enabled

**Test USER_FILE_NO_QRESOLUTION**
1. Start ISIS GUI and set the isntrument to SANS2DTUBES
2. Load the user file USER_FILE_NO_QRESOLUTION.txt (make sure that QResolution checkbox is not checked)
3. Load the data set SANS2D00028797.nxs as the sample data set
4. Run the 1D Reduce
   - Confirm that there are no errors.
5. Right click on the resulting workspace 28797rear_1D_1.75_16.5, select Show Data and press OK
   - Confirm that there is **no** tab with xerrors.  

**Test USER_FILE_QRESOLUTION_OFF.txt:**
1. Start ISIS GUI and set the instrument to SANS2DTUBES
2. Load the user file USER_FILE_QRESOLUTION_OFF.txt
3. Load the data set SANS2D00028797.nxs as the sample data set
4. Run the 1D Reduce
   - Confirm that there are no errors.
5. Right click on the resulting workspace 28797rear_1D_1.75_16.5, select Show Data and press OK
   - Confirm that there is **no** tab with xerrors.  

**Test USER_FILE_QRESOLUTION_CIRCULAR.txt**
1. Start ISIS GUI and set the isntrument to SANS2DTUBES
2. Load the user file USER_FILE_QRESOLUTION_CIRCULAR.txt
   - Confirm that Circ is selected, A1 and A2 are set, Moderator is set, LCollim and DeltaR are set
3. Load the data set SANS2D00028797.nxs as the sample data set
4. Run the 1D Reduce
   - Confirm that there are no errors.
5. Right click on the resulting workspace 28797rear_1D_1.75_16.5, select Show Data and press OK
   - Confirm that there **is**  tab with xerrors.  

**Test USER_FILE_QRESOLUTION_RECTANGULAR.txt**
1. Start ISIS GUI and set the isntrument to SANS2DTUBES
2. Load the user file USER_FILE_QRESOLUTION_RECTANGULAR.txt
   - Confirm that Circ is selected, H1 and H2 are set, Moderator is set, LCollim and DeltaR are set
3. Load the data set SANS2D00028797.nxs as the sample data set
4. Run the 1D Reduce
   - Confirm that there are no errors.
5. Right click on the resulting workspace 28797rear_1D_1.75_16.5, select Show Data and press OK
   - Confirm that there **is**  tab with xerrors. 

**Test USER_FILE_QRESOLUTION_INCOMPLETE.txt**
1. Start ISIS GUI and set the isntrument to SANS2DTUBES
2. Load the user file Test USER_FILE_QRESOLUTION_INCOMPLETE.txt. A2 is not set in this case
   - Confirm that the an error occurs with a message indicating that A2 has not been set.

**Test USER_FILE_QRESOLUTION_BAD_MODERATOR.txt**
1. Start ISIS GUI and set the isntrument to SANS2DTUBES
2. Load the user file Test USER_FILE_QRESOLUTION_BAD_MODERATOR.txt. Moderator file path is absent in this case
   - Confirm that the an error occurs with a message indicating that the moderator file is absent.

**Test USER_FILE_QRESOLUTION_PART.txt**
1. Start ISIS GUI and set the isntrument to SANS2DTUBES
2. Load the user file Test USER_FILE_QRESOLUTION_PART.txt. Info for rectangular is not complete, but for circualar is, hence circular should be selected.
   - On the reduction settings tab confirm that Circ. is selected.
3. Load the data set SANS2D00028797.nxs as the sample data set
4. Run the 1D Reduce
   - Confirm that there are no errors.
5. Right click on the resulting workspace 28797rear_1D_1.75_16.5, select Show Data and press OK
   - Confirm that there **is**  tab with xerrors.
